backend/api/v1/auth_service.go

```diff
@@ -311,45 +313,133 @@ func convertUserRole(userRole v1pb.UserRole) api.Role {
 	return api.UnknownRole
 }
 
-// Login is the auth login method.
+// Login is the auth login method including SSO.
 func (s *AuthService) Login(ctx context.Context, request *v1pb.LoginRequest) (*v1pb.LoginResponse, error) {
-	user, err := s.store.GetUserByEmail(ctx, request.Email)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to get principal by email %q", request.Email)
-	}
-	if user == nil {
-		return nil, status.Errorf(codes.Unauthenticated, "user %q not found", request.Email)
-	}
-	if user.MemberDeleted {
-		return nil, status.Errorf(codes.Unauthenticated, "user %q has been deactivated by administrators", request.Email)
+	var loginUser *store.UserMessage
+	if request.IdpName == "" {
```

Comment:
nit: split this large function into two smaller ones.

Reply:
Done